Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -355,43 +355,49 @@ void BackForwardCache::setMaxSize(unsigned maxSize)
     prune(PruningReason::None);
 }
 
-unsigned BackForwardCache::frameCount() const
+unsigned BackForwardCache::frameCount()
 {
     unsigned frameCount = m_items.size();
-    for (auto& item : m_items) {
-        ASSERT(item->m_cachedPage);
-        frameCount += item->m_cachedPage->cachedMainFrame()->descendantFrameCount();
+    for (auto& item : m_cachedPageMap) {
+        if (auto* cachedPage = std::get_if<UniqueRef<CachedPage>>(&item.value))
+            frameCount += (*cachedPage)->cachedMainFrame()->descendantFrameCount();
     }
     
     return frameCount;
 }
 
 void BackForwardCache::markPagesForDeviceOrPageScaleChanged(Page& page)
 {
-    for (auto& item : m_items) {
-        CheckedRef cachedPage = *item->m_cachedPage;
+    for (auto& item : m_cachedPageMap) {
+        auto* cachedPage = std::get_if<UniqueRef<CachedPage>>(&item.value);
+        if (!cachedPage)
```

Comment:
These should have at least an ASSERT_NOT_REACHED, or maybe a holds_alternative assertion.  Before, we used to just assume the cached page was non-null.  Now the assumption is in a different form but it should still let us know in a debug build if we break that assumption.

Reply:
That assertion won’t be true. The previous code assumed non-null values because it was iterating through `m_items`. Now, we iterate through `m_cachedPageMap`, which may contain identifiers that have been removed from `m_items` and instead store a pruning reason in the map.

Reply:
I added assertions verifying the identifier can’t be in `m_items` if the cached page is null here.